pkg/models/model.go

```diff
@@ -141,80 +129,28 @@ func (m *Model) setNodeMap() error {
 	return nil
 }
 
-func getNetworkLayers(name string, swmap map[string]string) ([]string, error) {
-	sw := make(map[string]bool)
-	res := []string{}
+func getNetworkLayers(sw *Switch, swmap map[string]*Switch) ([]string, map[string]string, error) {
+	visited := make(map[string]bool)
```

Comment:
Nit: For `visited`, you need a Set[string] so this could be `make(map[string]struct{})`